icsc2023_abstract_figure.py

```diff
@@ -4,9 +4,59 @@
 import numpy as np
 import matplotlib.pyplot as plt
 
-from simulate import (rr, rf, c_af, c_ar, c_maf, c_mar, c_pf, c_pr, ps, p_vals,
-                      p_arr, setup_initial_conditions, simulate,
-                      calc_linear_tire_force, eval_angles, plot_minimal)
+from simulate import *
+
+# batavus browser with Jason sitting on it, tire parameters from
+# Andrew/Gabriele
+p_vals = {
+    c_af: 11.46,  # estimates from Andrew's dissertation (done by him)
+    c_ar: 11.46,
+    c_f: 4000.0,  # guess
+    c_maf: 0.33,  # 0.33 is rough calc from Gabriele's data
+    c_mar: 0.33,
+    c_mpf: 0.0,  # need real numbers for this
+    c_mpr: 0.0,  # need real numbers for this
+    c_pf: 0.573,
+    c_pr: 0.573,
+    c_r: 4000.0,  # guess
+    d1: 0.9631492634872098,
+    d2: 0.4338396131640938,
+    d3: 0.0705000000001252,
+    g: 9.81,
+    ic11: 11.519805885486146,
+    ic22: 12.2177848012,
+    ic31: 1.57915608541552,
+    ic33: 2.959474124693854,
+    id11: 0.0883819364527,
+    id22: 0.152467620286,
+    ie11: 0.2811355367159554,
+    ie22: 0.246138810935,
+    ie31: 0.0063377219110826045,
+    ie33: 0.06782113764394461,
+    if11: 0.0904106601579,
+    if22: 0.149389340425,
+    k_f: 120000.0,  # ~ twice the stiffness of a 1.25" tire from Rothhamel 2024
+    k_r: 120000.0,  # ~ twice the stiffness of a 1.25" tire from Rothhamel 2024
+    l1: 0.5384415640161426,
+    l2: -0.531720230353059,
+    l3: -0.07654646159268344,
+    l4: -0.47166687226492093,
+    mc: 81.86,
+    md: 3.11,
+    me: 3.22,
+    mf: 2.02,
+    r_tf: 0.01,
+    r_tr: 0.01,
+    rf: 0.34352982332,
+    rr: 0.340958858855,
+    s_yf: 0.175,  # Andrew's estimates from his dissertation data
```

Comment:
I am testing with 50% relaxation length. I didn't measure it for this tyre, however I believe it is more reasonable to use 50% of this value

Reply:
That's fine, but do you think I have a correct simulation here? Changing that parameter 50% will make some small differences, but more importantly if we have a valid model.

Reply:
The model looks generally valid, but it is still hard to have simulations close to the reality. To obtain roll rate and yaw rate in the same order of magnitude of experimental results, I need to push the acceleration up to >100 m/s^2

Reply:
To have simulations close to reality, we would need to obtain roll rate and yaw rate ~100 deg/s. Now, we obtain ~60 deg/s, with kickplate acceleration > 100 m/s^2.

Reply:
100 m/s/s will cause extremely large slip angles (unrealistic ones, right?).

Reply:
Yes, right. We have slip angles up to 25 deg, quite unrealistic. To have realistic slip angles (~10 deg) and realistic kickplate acceleration (~40-50 m/s/s), we have to accept yaw rate and roll rate ~25 deg/s (against ~100-120 deg/s of the experiment). 
I thought an error in the tyre model: the tyre is not able to generate such large forces with "small" (~10 deg) slip angles. Thus the model tries to find larger values of lateral force by pushing the slip angle to unrealistic limits. I simulated with lateral force multiplied by 10 (exaggerated and unrealistic, only to understand), but the simulations broke.

Reply:
I am not sure we can get any better than that right now.